mantle/cmd/kolet/kolet.go

```diff
@@ -138,9 +139,16 @@ func dispatchRunExtUnit(unitname string, sdconn *systemddbus.Conn) (bool, error)
 					}
 				case CLD_KILLED:
 					// SIGTERM; we explicitly allow that, expecting we're rebooting.
-					if mainstatus == 15 {
+					if mainstatus == int32(15) {
 						fmt.Printf("Unit %s terminated via SIGTERM, assuming reboot\n", unitname)
-						return false, nil
+						// Now we send ourself SIGTERM, so that that propagates back to the ssh
+						// status, so that the kola runner can use that as a trigger to reboot.
+						selfproc := os.Process{
+							Pid: os.Getpid(),
+						}
+						selfproc.Signal(syscall.SIGTERM)
+						time.Sleep(time.Hour)
```

Comment:
It feels odd to wait wait up to an hour, but then again it won't hurt to wait that long since it should be interrupted.